app/services/preserver.rb

```diff
@@ -31,21 +31,38 @@ def preserve!
     end
   end
 
-  def preserve_original_file
-    return unless resource.try(:original_file) && preservation_object.binary_nodes.blank?
-    file_metadata = FileMetadata.new(
-      label: preservation_copy_label,
+  def preserve_binary_content
+    binary_nodes.each do |binary_node|
+      next if binary_node.file_identifiers.blank?
+      next if preservation_object.binary_nodes.find { |x| x.preservation_copy_of_id == binary_node.id }
```

Comment:
Shouldn't this be `next unless` to skip binaries that aren't marked as the file to be preserved?

Reply:
No, it's `next if` so it won't re-upload binaries that it's already uploaded.

Reply:
I see the test verifying that both the preservation master and intermediate are saved, so I think I am just misreading this: https://github.com/pulibrary/figgy/pull/2821/commits/40f70fe1e4ee75a6a591200e079f6567f14b061f#diff-2361c599e97a22f4ee2ebef0aba33d40R1557

Reply:
Maybe it's the variable names? Want me to rename it to `resource_binary_nodes` to differentiate from `preservation_object.binary_nodes`?

Reply:
No, I think it's fine — I see how it's working now.